fishtank/src/cluster.ts

```diff
@@ -51,20 +57,53 @@ export class Cluster {
       labels: { [CLUSTER_LABEL]: this.name },
     }
 
+    const args: string[] = []
+
     if (options.config) {
       const configString = JSON.stringify(options.config)
 
-      const dest = join(tmpdir(), 'fishtank', containerName, '.ironfish')
+      const dest = join(tmpdir(), 'fishtank', this.name, options.name, '.ironfish')
       await promises.mkdir(dest, {
         recursive: true,
       })
 
       await promises.writeFile(resolve(dest, 'config.json'), configString)
 
-      runOptions.volumes = new Map<string, string>([[dest, '/root/.ironfish']])
+      if (runOptions.volumes === undefined) {
```

Comment:
Initialize `runOptions` with an empty `Map` so that you don't have to check for `undefined` every time:

```typescript
const runOptions: RunOptions = { 
  name: containerName,
  networks: [this.networkName()],
  hostname: options.name,
  labels: { [CLUSTER_LABEL]: this.name },
  volumes: new Map(),
}

runOptions.volumes.set(dest, DEFAULT_CONTAINER_DATADIR) // no undefined check required
runOptions.volumes.set(dest, containerPath) // no undefined check required
```

Reply:
Still need to check runOptions.volumes because the type is undefined|map. Unless the type is changed to  volume: map, not allow it to be optional.